adapters/oidc/spring-security/src/main/java/org/keycloak/adapters/springsecurity/authentication/KeycloakAuthenticationEntryPoint.java

```diff
@@ -94,7 +94,17 @@ public void commence(HttpServletRequest request, HttpServletResponse response, A
         }
     }
 
+    /**
+     * Redirects to the login page. If HTTP sessions are disabled, the redirect URL is saved in a
+     * cookie now, to be retrieved by the {@link KeycloakAuthenticationSuccessHandler} or the
+     * {@link KeycloakAuthenticationFailureHandler} when the login sequence completes.
+     */
     protected void commenceLoginRedirect(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        if (request.getSession(false) == null && KeycloakCookieBasedRedirect.getRedirectUrlFromCookie(request) == null) {
+            // If no session exists yet at this point, then apparently the redirect URL is not
+            // stored in a session. We'll store it in a cookie instead.
+            response.addCookie(KeycloakCookieBasedRedirect.createCookieFromRedirectUrl(request.getRequestURI()));
```

Comment:
This does not preserve query parameters of the original request. Is this intentional? Is there any (e.g. security related) reason to not include the query?

Reply:
Not intenional, this is an oversight on my part. I think it would make perfect sense to include the query parameters.

Reply:
> If the current Keycloak adapter already has an endpoint that finalises the authentication flow which accepts a redirect URI parameter, then maybe we could reuse that and avoid the extra cookie (which is a burden because of the HTTP headers growing rather large)

Sounds good in principle, but I am not sure if that is the case. I wouldn't worry too much about it though, because this additional cookie is so much smaller than the ADAPTER_STATE cookie. If header size is a concern, this entire approach isn't ideal.